docs/source/coingecko.rst

```diff
@@ -12,11 +12,11 @@ Given base token name and the support currencies to get token price from CoinGec
 **Parameters**
 
 - ``base: string`` - the token name that you can get at URL while search in CoinGecko. Or find the token from https://api.coingecko.com/api/v3/coins/list
-- ``quote: string`` - the support currencies in CoinGecko, see https://api.coingecko.com/api/v3/simple/supported_vs_currencies
+- ``quote: string`` - (optional) the support currencies in CoinGecko. Default quote is `usd`. See https://api.coingecko.com/api/v3/simple/supported_vs_currencies
 
 **Returns**
 
-- ``number`` - token price
+- ``number | undefined`` - token price
```

Comment:
Should this return number or BigNumber or perhaps a string?
Since javascript number has its limits

Reply:
I'm not worried that since the token price is usually present in usd

Reply:
Is USD the only quote currency supported?
I guess not, some might put the quote currency as ethnor other 18 decimal currencies

Reply:
you are right, checked and found the support quotes includes `btc`, which might have the digits issue https://api.coingecko.com/api/v3/simple/supported_vs_currencies

update to return string instead of number